File: samples/client/3_1_0_unit_test/java/src/test/java/org/openapijsonschematools/client/components/schemas/TypeArrayOrObjectTest.java
package org.openapijsonschematools.client.components.schemas;

import org.junit.Assert;
import org.junit.Test;
import org.openapijsonschematools.client.configurations.JsonSchemaKeywordFlags;
import org.openapijsonschematools.client.configurations.SchemaConfiguration;
import org.openapijsonschematools.client.exceptions.ValidationException;
import org.openapijsonschematools.client.schemas.validation.MapUtils;
import org.checkerframework.checker.nullness.qual.Nullable;

import java.util.Arrays;
import java.util.List;
import java.util.Map;
import java.util.AbstractMap;

public class TypeArrayOrObjectTest {
    static final SchemaConfiguration configuration = new SchemaConfiguration(new JsonSchemaKeywordFlags.Builder().format().build());

    @Test
    public void testNumberIsInvalidFails() {
        // number is invalid
        final var schema = TypeArrayOrObject.TypeArrayOrObject1.getInstance();
        try {
            schema.validate(
                123,
                configuration
            );
            throw new RuntimeException("A different exception must be thrown");
        } catch (ValidationException ignored) {
            ;
        }
    }

    @Test
    public void testStringIsInvalidFails() {
        // string is invalid
        final var schema = TypeArrayOrObject.TypeArrayOrObject1.getInstance();
        try {
            schema.validate(
                "foo",
                configuration
            );
            throw new RuntimeException("A different exception must be thrown");
        } catch (ValidationException ignored) {
            ;
        }
    }

    @Test
    public void testNullIsInvalidFails() {
        // null is invalid
        final var schema = TypeArrayOrObject.TypeArrayOrObject1.getInstance();
        try {
            schema.validate(
                (Void) null,
                configuration
            );
            throw new RuntimeException("A different exception must be thrown");
        } catch (ValidationException ignored) {
            ;
        }
    }

    @Test
    public void testArrayIsValidPasses() throws ValidationException {
        // array is valid
        final var schema = TypeArrayOrObject.TypeArrayOrObject1.getInstance();
        schema.validate(
            Arrays.asList(
                1,
                2,
                3
            ),
            configuration
        );
    }

    @Test
    public void testObjectIsValidPasses() throws ValidationException {
        // object is valid
        final var schema = TypeArrayOrObject.TypeArrayOrObject1.getInstance();
        schema.validate(
            MapUtils.makeMap(
                new AbstractMap.SimpleEntry<String, Number>(
                    "foo",
                    123
                )
            ),
            configuration
        );
    }
}

File: samples/client/3_1_0_unit_test/kotlin/src/test/kotlin/org/openapijsonschematools/client/components/schemas/TypeArrayOrObjectTest.kt
package org.openapijsonschematools.client.components.schemas

import org.openapijsonschematools.client.configurations.JsonSchemaKeyword
import org.openapijsonschematools.client.configurations.SchemaConfiguration
import org.openapijsonschematools.client.exceptions.ValidationException

import kotlin.test.Test
import kotlin.test.assertFailsWith

class TypeArrayOrObjectTest {
    companion object {
        val configuration = SchemaConfiguration(disabledKeywordFlags=setOf(JsonSchemaKeyword.FORMAT))
    }

    @Test
    fun testNumberIsInvalidFails() {
        // number is invalid
        val schema = TypeArrayOrObject.TypeArrayOrObject1.getInstance()
        assertFailsWith<ValidationException>(
            block = {
                schema.validate(
                    123,
                    configuration
                )
            }
        )
    }

    @Test
    fun testStringIsInvalidFails() {
        // string is invalid
        val schema = TypeArrayOrObject.TypeArrayOrObject1.getInstance()
        assertFailsWith<ValidationException>(
            block = {
                schema.validate(
                    "foo",
                    configuration
                )
            }
        )
    }

    @Test
    fun testNullIsInvalidFails() {
        // null is invalid
        val schema = TypeArrayOrObject.TypeArrayOrObject1.getInstance()
        assertFailsWith<ValidationException>(
            block = {
                schema.validate(
                    null,
                    configuration
                )
            }
        )
    }

    @Test
    fun testArrayIsValidPasses() {
        // array is valid
        val schema = TypeArrayOrObject.TypeArrayOrObject1.getInstance()
        schema.validate(
            listOf(
                1,
                2,
                3
            ),
            configuration
        )
    }

    @Test
    fun testObjectIsValidPasses() {
        // object is valid
        val schema = TypeArrayOrObject.TypeArrayOrObject1.getInstance()
        schema.validate(
            mapOf(
                Pair(
                    "foo",
                    123
                )
            ),
            configuration
        )
    }
}
